src/Resources/ResourceManager/Extensions/JTokenExtensions.cs

```diff
@@ -123,5 +123,91 @@ internal static object ConvertPropertyValueForPsObject(JToken propertyValue)
 
             return propertyValue.ToString();
         }
+
```

Comment:
These two new methods are similar enough to the existing methods, it's worth looking for ways to reduce the duplication. For example, adding an optional boolean parameter to the ToPsObject method:

```ToPsObject(this JToken jtoken, string objectType = null, bool deepConvert = false)```

captures the semantics nicely without adding a method or breaking any existing uses of the current method. To implement it you can add that parameter and a conditional block to the existing ConvertPropertyValueForPsObject method like this:
```
        internal static object ConvertPropertyValueForPsObject(this JToken propertyValue, bool deepConvert)
        {
            if (propertyValue.Type == JTokenType.Object)
            {
                return propertyValue.ToPsObject(deepConvert: deepConvert);
            }

            if (propertyValue.Type == JTokenType.Array)
            {
                var jArray = (JArray)propertyValue;

                var array = new object[jArray.Count];

                for (int i = 0; i < array.Length; ++i)
                {
                    array[i] = jArray[i].ConvertPropertyValueForPsObject(deepConvert);
                }

                return array;
            }

>>>
            if (deepConvert && propertyValue is JProperty property)
            {
                return property.Value.ToPsObject(deepConvert: deepConvert);
            }
>>>

            Type primitiveType;
            if (JTokenExtensions.PrimitiveTypeMap.TryGetValue(propertyValue.Type, out primitiveType))
            {
                try
                {
                    return propertyValue.ToObject(primitiveType, JsonExtensions.JsonObjectTypeSerializer);
                }
                catch (FormatException)
                {
                }
                catch (ArgumentException)
                {
                }
                catch (JsonException)
                {
                }
            }

            return propertyValue.ToString();
        }
```

Note, I also changed it to an extension method for consistency.

I don't know for sure if this builds out the properties exactly the same way the two new methods did, but it's effectively the same and the details should be pretty easy to tweak. I think this idea is worth looking at: it also fixes the other issue in which one of the conversion paths doesn't do deep conversion.